frontend/src/theme/utils.ts

```diff
@@ -412,10 +478,12 @@ export const createEmotionTheme = (
     genericFonts: {
       ...genericFonts,
       ...(parsedFont && {
-        bodyFont: parsedFont,
-        headingFont: parsedFont,
+        bodyFont: themeInput.bodyFont ? themeInput.bodyFont : parsedFont,
+        headingFont: themeInput.bodyFont ? themeInput.bodyFont : parsedFont,
+        codeFont: themeInput.codeFont ? themeInput.codeFont : parsedFont,
```

Comment:
@sfc-gh-tteixeira I am not sure if we want the same font in 3 places. This probably caused the regressions. See: 
https://github.com/streamlit/streamlit/issues/6484